src/qonnx/transformation/quantize_graph.py

```diff
@@ -0,0 +1,261 @@
+                    consumer of the input to the quant node.
+
+        3) Introduction to quantnodes will be done with precedence to "Name" in comparison to "op_type".
+
+        4) Assert:
+            a) The input is a dictionary representing the node names as keys and a list of quant positions
+                as values.
+            b) The input dictionary must have atleast one mac node (Conv, gemm, matmul) for the transformation.
+
+        5) Return:
+            Returns a cleaned version of the model.
+
+    """
+
+    def __init__(self, quant_node_inputs):
+        super().__init__()
+        self.quant_node_inputs = quant_node_inputs
+        self.graph_util = graph_util()
+
+    def apply(self, model):
+        model = model.transform(InferShapes())
+        if type(self.quant_node_inputs) == dict:
+            selection_type = self.quant_node_inputs.keys()
+            if set(selection_type) <= {"name", "op_type"}:
+                node_count = 0
+                quantized_nodes = []
+                if "name" in selection_type:
+                    by_name = self.quant_node_inputs[
+                        "name"
+                    ]  # by_name is a dictionary with the unique node names as keys and the list of positions as values.
+                    node_list_by_name = by_name.keys()  # name of all the nodes specified by the user for an quant node.
+                    for node_name in node_list_by_name:
+                        node_in_focus = self.graph_util.node_from_name(model, node_name)
+                        input_positions = by_name[
+                            node_name
+                        ]  # input positions specified by the user to introduce quant node.
+                        model = self.graph_util.adjust_graph(
+                            model, input_positions, node_in_focus, quantized_nodes, node_count
+                        )
+                        node_count += 1
+                if "op_type" in selection_type:
+                    by_op_type = self.quant_node_inputs[
+                        "op_type"
+                    ]  # by_name is a dictionary with the unique node names as keys and the list of positions as values.
+                    op_list = by_op_type.keys()
+                    for op in op_list:
+                        node_list = self.graph_util.identify_nodes(
+                            model, op
+                        )  # List of all nodes with the operation type "op".
+                        input_positions = by_op_type[op]
+                        for node_in_focus in node_list:
+                            model = self.graph_util.adjust_graph(
+                                model, input_positions, node_in_focus, quantized_nodes, node_count
+                            )
+                            node_count += 1
+                model = qonnx_make_model(model.graph)
+                model = ModelWrapper(model)
```

Comment:
you should be able to just remove these two lines, they are redundant

Reply:
"redundant" lines are removed in the latest update.